src/Files.App.Server/Utils/RegistryUtils.cs

```diff
@@ -0,0 +1,204 @@
+﻿using Microsoft.Win32;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Files.App.Server.Utils
+{
+	internal sealed class RegistryUtils
+	{
+		internal static string CombineKeys(string key1, string key2)
+		{
+			key1 = key1.Replace('/', '\\').TrimStart('\\').TrimEnd('\\');
+			key2 = key2.Replace('/', '\\').TrimStart('\\').TrimEnd('\\');
+			return $"{key1}\\{key2}";
+		}
+
+		internal static string CombineKeys(string key1, string key2, string key3)
+		{
+			key1 = key1.Replace('/', '\\').TrimStart('\\').TrimEnd('\\');
+			key2 = key2.Replace('/', '\\').TrimStart('\\').TrimEnd('\\');
+			key3 = key3.Replace('/', '\\').TrimStart('\\').TrimEnd('\\');
+			return $"{key1}\\{key2}\\{key3}";
+		}
+
+		internal static void BindValues(RegistryKey key, object? target, string namePrefix = "")
+		{
+			if (target is null)
+			{
+				return;
+			}
+
+			foreach (var p in target.GetType().GetProperties(BindingFlags.Public | BindingFlags.Instance))
+			{
+				if ((p.PropertyType == typeof(string)
+					|| p.PropertyType == typeof(string[])
+					|| p.PropertyType.IsEnum
+					|| p.PropertyType.IsPrimitive
+					|| (p.PropertyType.IsGenericType && p.PropertyType.GetGenericTypeDefinition() == typeof(Nullable<>)))
```

Comment:
Even with reflection are you sure it's faster? Can we cache the Reflection somewhere like a `FrozenDictionary`?

Reply:
The engine of LiteDB uses reflection and expression tree to build the query, so we are just moving from reflection to another reflection here.

Reply:
Ideally I would like to use source generator to generate code so that we can avoid reflection at all.

Reply:
Fair, I think we can do this is another scope, solving the crash is more important :)

Reply:
I have switched to use a source generator for this.

Reply:
Congrats, looks awesome!